File: src/main/java/kr/jongyeol/jsBot/data/ModRoles.java
package kr.jongyeol.jsBot.data;

import lombok.Data;

@Data
public class ModRoles {
    private long releaseRole = -1;
    private long progressRole = -1;
    private long betaReleaseRole = -1;

    public boolean notSet() {
        return releaseRole == -1 && progressRole == -1 && betaReleaseRole == -1;
    }

    public String getReleasePing() {
        return "<@&" + releaseRole + ">";
    }

    public String getProgressPing() {
        return "<@&" + progressRole + ">";
    }

    public String getBetaReleasePing() {
        return "<@&" + betaReleaseRole + ">";
    }
}

File: JsBot/Data/ModRoles.cs
using System.Text.Json;

namespace JsBot.Data;

public class ModRoles {
	public long ReleaseRole = -1;
	public long ProgressRole = -1;
	public long BetaReleaseRole = -1;

	public bool NotSet() => ReleaseRole == -1 && ProgressRole == -1 && BetaReleaseRole == -1;

	public string ReleasePing => "<@&" + ReleaseRole + ">";
	public string ProgressPing => "<@&" + ProgressRole + ">";
	public string BetaReleasePing => "<@&" + BetaReleaseRole + ">";

	public void Load(ref Utf8JsonReader reader) {
		while(reader.Read() && reader.TokenType != JsonTokenType.EndObject) {
			if(reader.TokenType != JsonTokenType.PropertyName) continue;
			string propertyName = reader.GetString()!;
			reader.Read();
			switch(propertyName) {
				case "releaseRole":
					ReleaseRole = reader.GetInt64();
					break;
				case "progressRole":
					ProgressRole = reader.GetInt64();
					break;
				case "betaReleaseRole":
					BetaReleaseRole = reader.GetInt64();
					break;
				default:
					reader.Skip();
					break;
			}
		}
	}

	public void Save(Utf8JsonWriter writer) {
		writer.WriteStartObject();
		writer.WriteNumber("releaseRole", ReleaseRole);
		writer.WriteNumber("progressRole", ProgressRole);
		writer.WriteNumber("betaReleaseRole", BetaReleaseRole);
		writer.WriteEndObject();
	}
}
